e2e_testing/torchscript/xfail_sets.py

```diff
@@ -154,3 +154,312 @@
     "TestMultipleTensorReturn_basic",
     "AdaptiveAvgPool2dUnitOutputSizeStaticModule_basic",
 }
+
+LTC_XFAIL_SET = {
+    "AdaptiveAvgPool2dNonUnitOutputSizeDynamicModule_basic",
```

Comment:
Any triage on why this could be failing?

If we are running the JIT graph directly, I would assume that everything "Just Works".

Reply:
We don't have support for all ops yet. Some are missing some shape inference functions, some ops are blacklisted, etc.

Reply:
Ah, got it. It's a little annoying that we are duplicating the shape functions for LTC. Could you collaborate with @eellison to reuse their shape stuff?

We have been upstreaming all our shape functions from https://github.com/llvm/torch-mlir/blob/main/python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/shape_lib_gen.py, so this should all be part of a big source of truth upstream.

Reply:
All of the shapes functions are also stored in C++ fwiw, so I don't think that's a fundamental blocker.  but I dont know all the details of what's going on here

Reply:
@eellison Can you point me to where that is in the PyTorch repo? I just looked at the PR that Sean linked earlier, which had C++ with some long strings.

Reply:
It should turn into JIT IR (that's what you see in the strings) that we can then load up and execute from C++.

Reply:
@silvasean are you suggesting that we apply this at the JIT graph level, or port this over to work with LTC generally?

Reply:
If it's the former, I don't _think_ we can use it, because we require shape information at the LTC layer before generating a JIT graph.

Reply:
Oh, interesting. Yeah, that makes things more tricky to reuse. @eellison is the shape inference infra tied specifically to JIT graphs?

I suppose we could create a shapeless jit graph just for the purpose of reusing the shape inference infrastructure, but that's quite a few hoops to jump through.